libs/SalesforceSDK/src/com/salesforce/androidsdk/config/RuntimeConfig.java

```diff
@@ -108,5 +110,12 @@ private Bundle getRestrictions(Context ctx) {
 		RestrictionsManager restrictionsManager = (RestrictionsManager) ctx.getSystemService(Context.RESTRICTIONS_SERVICE);
 		return restrictionsManager.getApplicationRestrictions();
 	}
-		
+
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private boolean hasRestrictionsProvider(Context ctx) {
+        RestrictionsManager restrictionsManager = (RestrictionsManager) ctx.getSystemService(Context.RESTRICTIONS_SERVICE);
+        return restrictionsManager.hasRestrictionsProvider();
```

Comment:
This is the way to check whether the application is managed. On lollipop, the restrictions bundle is not null whether the app is managed or not.